backtrace/src/tasks.rs

```diff
@@ -27,12 +28,29 @@ pub(crate) fn deregister(root_frame: &Frame) {
 }
 
 /// An iterator over tasks.
-pub(crate) fn tasks() -> impl Iterator<Item = impl Deref<Target = Task>> {
+///
+/// **NOTE:** The creation and destruction of some or all tasks will be blocked
+/// for as long as the return value of this function is live.
+pub fn tasks() -> impl Iterator<Item = impl Deref<Target = Task>> {
     TASK_SET.iter()
 }
 
 impl Task {
-    pub(crate) fn dump_tree(&self, wait_for_running_tasks: bool) -> String {
+    /// The location of this task.
+    pub fn location(&self) -> crate::Location {
+        // safety: we promise to not inspect the subframes without first locking
+        let frame = unsafe { self.0.as_ref() };
+        frame.location()
+    }
+
+    /// Pretty-prints this task as a tree.
+    ///
+    /// If `block_until_idle` is `false`, the output will note that this task is
```

Comment:
This is only true if the task is being polled.  Also, I think it would also be good to include a note about the possibility of deadlock if `block=true`.  Here's an alternate phrasing:

// If `block_until_idle` is `true`, this routine will block until the task is no longer being polled.  In this case, the caller should not hold any locks which might be held by the task, otherwise deadlock may occur.
//
// If `block_until_idle` is `false`, and the task is being polled, the output will not include the sub-frames, instead simply note that the task is being polled.

Another approach would be to have 2 `pub fn`'s, `pretty_tree(&self) -> String`, and `pretty_tree_blocking(&self) -> String`.  This would make it more clear that the nonblocking use is preferred (due to performance and deadlock potential).